src/coreneuron/io/nrn_setup.cpp

```diff
@@ -216,6 +216,12 @@ void nrn_read_filesdat(int& ngrp, int*& grp, const char* filesdat) {
 
         nrn_assert(fscanf(fp, "%d\n", &iFile) == 1);
         if ((iNum % nrnmpi_numprocs) == nrnmpi_myid) {
+            // A "-1" entry means that this rank should not be assigned further gid groups.
+            // It is a way to create files.dat files which deterministically assign gid groups to
+            // ranks, particularly useful for very large simulations which required load balancing.
+            if (iFile == -1) {
```

Comment:
@nrnhines : this is a bit of BBP specific change where we rebalance the `files.dat` with some extra sauce (i.e. we modify `files.dat` from external python program). I can explain details (offline) but this doesn't change anything for normal neuron executuon.

Could you just confirm if files.dat could have an entry with `-1` as a gid? I don't remember very well and hence want to double check with you.

Reply:
I'm fine with this assuming that NEURON is not writing the files.dat with a negative gid.